### PR TITLE
Make context menus in scene editor and in object list consistent

### PR DESCRIPTION
**Pull request by AlexandreSi**
- Right-click on instance:

  image

- Right-click on background (even if instance or object is selected):

  image

- Right-click on object:
  - Action `Add instance to the scene` adds an instance at the center of the scene

  image

- Duplicate action on instances adds duplicates on the lower right side, at the distance of 2 shift+arrow moves:
   
  image

### CODE REVIEW

File: newIDE/app/src/SceneEditor/index.js [hunk 1]
```diff
@@ -850,8 +863,127 @@ export default class SceneEditor extends React.Component<Props, State> {
...
-    if (this.contextMenu) this.contextMenu.open(x, y);
+  _onContextMenu = (
+    x: number,
+    y: number,
+    ignoreSelectedObjectNamesForContextMenu?: boolean = false
+  ) => {
+    if (!ignoreSelectedObjectNamesForContextMenu) {
+      if (this.contextMenu) this.contextMenu.open(x, y);
+    } else {
+      this.setState({ ignoreSelectedObjectNamesForContextMenu: true }, () => {
```
Comment (4ian) [inline newIDE/app/src/SceneEditor/index.js:874]: a bit "dirty", I wonder if we should allow some arbitrary parameters to be passed as third option to `contextMenu.open` that would be then passed to buildMenuTemplate? :)

File: newIDE/app/src/SceneEditor/index.js [hunk 2]
```diff
@@ -876,6 +1008,33 @@ export default class SceneEditor extends React.Component<Props, State> {
...
+
+    if (!this.editor) return;
+
+    const newInstances = serializedSelection
+      .map(serializedInstance => {
+        const instance = new gd.InitialInstance();
+        unserializeFromJSObject(instance, serializedInstance);
+        return instance;
+      })
+      .map(instance => {
```
Comment (4ian) [inline newIDE/app/src/SceneEditor/index.js:1024]: Do you need to do two maps? If you do a single one, this allows to keep the new in the same function as the `delete`, which is looking a tiny bit safer for the reader.

File: newIDE/app/src/SceneEditor/index.js (hunk 1 above)
Comment (AlexandreSi) [reply newIDE/app/src/SceneEditor/index.js:874]: Yes it is dirty.

I thought about that as well, but I didn't want to change the signature for this often-used method. But this is a case for which it's worth I think.

File: newIDE/app/src/InstancesEditor/index.js [hunk 3]
```diff
@@ -604,20 +609,40 @@ export default class InstancesEditor extends Component<Props> {
...
   |}) => {
     this.lastContextMenuX = offsetX;
     this.lastContextMenuY = offsetY;
-    if (this.props.onContextMenu) this.props.onContextMenu(x, y);
+    if (this.props.onContextMenu) {
+      const _ignoreSelectedObjectNamesForContextMenu =
+        typeof ignoreSelectedObjectNamesForContextMenu === 'boolean'
+          ? ignoreSelectedObjectNamesForContextMenu
+          : false;
+      this.props.onContextMenu(x, y, _ignoreSelectedObjectNamesForContextMenu);
```
Comment (4ian) [inline newIDE/app/src/InstancesEditor/index.js:644]:
```suggestion
      this.props.onContextMenu(x, y, ignoreSelectedObjectNamesForContextMenu || false);
```
or 
```suggestion
      this.props.onContextMenu(x, y, !!ignoreSelectedObjectNamesForContextMenu);
```

Should work (undefined == falsy)

File: newIDE/app/src/SceneEditor/index.js (hunk 2 above)
Comment (4ian) [reply newIDE/app/src/SceneEditor/index.js:1024]: I believe you can also change the `paste` function to remove the two maps yeah, it seems useless..

**Review by 4ian — APPROVED** [review]
LGTM after resolving comments! :)

**Pull request merged**